modules/base/rendering/renderableprism.cpp

```diff
@@ -0,0 +1,354 @@
+        "Segments",
+        "The number of segments the shape of the prism should have."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LinesInfo = {
+        "NumLines",
+        "Number of Lines",
+        "The number of lines connecting the two shapes of the prism."
+    };
+
+    static const openspace::properties::Property::PropertyInfo RadiusInfo = {
+        "Radius",
+        "Radius",
+        "The radius of the prism's shape in meters."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineWidthInfo = {
+        "LineWidth",
+        "Line Width",
+        "This value specifies the line width."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineColorInfo = {
+        "Color",
+        "Color",
+        "This value determines the RGB color for the line."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LengthInfo = {
+        "Length",
+        "Length",
+        "The length of the prism in meters."
+    };
+
+    // Generate vertices around the unit circle on the XY-plane
+    void getUnitCircleVertices(std::vector<float>& vertices, int sectorCount) {
+        float sectorStep = 2 * M_PI / sectorCount;
+        float sectorAngle;  // in radians
+
+        for (int i = 0; i < sectorCount; ++i) {
+            sectorAngle = i * sectorStep;
+            vertices.push_back(cos(sectorAngle)); // x
+            vertices.push_back(sin(sectorAngle)); // y
+        }
+    }
+
+    struct [[codegen::Dictionary(RenderablePrism)]] Parameters {
+        // [[codegen::verbatim(SegmentsInfo.description)]]
+        int segments;
+
+        // [[codegen::verbatim(LinesInfo.description)]]
+        std::optional<int> lines;
+
+        // [[codegen::verbatim(RadiusInfo.description)]]
+        std::optional<float> radius;
+
+        // [[codegen::verbatim(LineWidthInfo.description)]]
+        std::optional<float> lineWidth;
+
+        // [[codegen::verbatim(LineColorInfo.description)]]
+        std::optional<glm::vec3> color [[codegen::color()]];
+
+        // [[codegen::verbatim(LengthInfo.description)]]
+        std::optional<float> length;
+    };
+#include "renderableprism_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderablePrism::Documentation() {
+    documentation::Documentation doc = codegen::doc<Parameters>("base_renderable_prism");
+    return doc;
+}
+
+RenderablePrism::RenderablePrism(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _nShapeSegments(SegmentsInfo, 6, 3, 32)
+    , _nLines(LinesInfo, 6, 0, 32)
+    , _radius(RadiusInfo, 10.f, 0.f, 3.0e12f)
+    , _lineWidth(LineWidthInfo, 1.f, 1.f, 20.f)
+    , _lineColor(LineColorInfo, glm::vec3(1.f), glm::vec3(0.f), glm::vec3(1.f))
+    , _length(LengthInfo, 20.f, 1.f, 3.0e12f)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _radius.setViewOption(properties::Property::ViewOptions::Logarithmic);
```

Comment:
@WeirdRubberDuck 

I don't think this is necessary anymore (or even compiles anymore) in the new `master` as you'd set the exponent of the slider now instead

Reply:
Yep, that is true. To get the same behavior as with the previously `Logarithmic` viewoption you can set the exponent to the tenth logarithm of the properties max value (i.e. here: `_radius.setExponent(12.f)`). However, I would suggest setting it to a slightly smaller one. Best is to set a value and try it out to see if it behaves the way you'd want it